src/test/java/core/basesyntax/service/ReaderServiceImplTest.java

```diff
@@ -0,0 +1,206 @@
+package core.basesyntax.service;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.junit.Assert;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+
+class ReaderServiceImplTest {
+    private static final String PATH_TEST = "src/test/resources/";
+    private static final String FILE_NAME = "fruits.csv";
+    private static final String FILE_EMPTY = "fruits_empty.csv";
+    private static final String FILE_NO_DATA = "fruits_no_data.csv";
+    private static final String FILE_BAD_OPERATION = "fruits_bad_operation.csv";
+    private static final String FILE_BAD_NUMBER = "fruits_bad_number.csv";
+    private static final String FILE_STORAGE = "fruits_storage.csv";
+    private static final File testFile = new File(PATH_TEST + FILE_NAME);
+    private static final File emptyFile = new File(PATH_TEST + FILE_EMPTY);
+    private static final File noDataFile = new File(PATH_TEST + FILE_NO_DATA);
+    private static final File badOperationFile = new File(PATH_TEST + FILE_BAD_OPERATION);
+    private static final File badNumberFile = new File(PATH_TEST + FILE_BAD_NUMBER);
+    private static final File storageFile = new File(PATH_TEST + FILE_STORAGE);
+    private static final String FIRST_LINE = "type,fruit,quantity";
+    private static final String NEW_LINE = System.lineSeparator();
```

Comment:
there is no need to make that many constants in test classes. that's the place where you can use hardcode